components/homme/utils/csm_share/CMakeLists.txt

```diff
@@ -0,0 +1,57 @@
+# Make sure we create the target once
+if (NOT TARGET csm_share)
+  set (SHARE_DIR ${HOMME_SOURCE_DIR}/../../share/util)
+
+  set (CSM_SHARE_SOURCES)
+
+  # First, handle the files that need to go through genf90
+
+  # Set variables needed for processing genf90 templates
+  set(CIMEROOT ${HOMME_SOURCE_DIR}/../../cime)
+  list(APPEND CMAKE_MODULE_PATH ${CIMEROOT}/CIME/non_py/src/CMake)
+  set(GENF90 ${CIMEROOT}/CIME/non_py/externals/genf90/genf90.pl)
+  set(ENABLE_GENF90 True)
+  include(genf90_utils)
+  include(Sourcelist_utils)
+
+  # GENF90_SOURCE lists source files we will need to run through the genf90 perl script
+  set (GENF90_SOURCE
+    ${SHARE_DIR}/shr_infnan_mod.F90.in
+  )
+
+  # Process genf90 template files. For each src file, this adds a custom command
+  # (and hence target) that needs to be built from the genf90 template files
+  foreach (SRC_FILE ${GENF90_SOURCE})
+    string(REPLACE ".in" "" SRC_FILE_STRIPPED ${SRC_FILE})
+    get_filename_component(BASENAME ${SRC_FILE_STRIPPED} NAME)
+    set(SRC_FILE_OUT "${CMAKE_CURRENT_BINARY_DIR}/${BASENAME}")
+    add_custom_command (
+      OUTPUT ${SRC_FILE_OUT}
+      COMMAND ${GENF90} ${SRC_FILE} > ${SRC_FILE_OUT}
+      DEPENDS ${SRC_FILE} genf90)
+    list(APPEND CSM_SHARE_SOURCES ${SRC_FILE_OUT})
+  endforeach ()
+
+  # "normal" sources
+  list (APPEND CSM_SHARE_SOURCES
+    ${SHARE_DIR}/shr_abort_mod.F90
```

Comment:
Note: there were more modules in the original csm_share folder, but turned out many of them were never used (directly or indirectly) in standalone homme. So I only included the ones that are effectively used.

If other modules end up being used in the future, we simply have to add them here.